### PR TITLE
feat: steps support for circular progress bar

### LINKED ISSUE #35802: Progress: circle模式下支持steps
### What problem does this feature solve?

Progress在现有circle模式下传入steps参数并不起效果
反而会使中间内容被遮盖

### What does the proposed API look like?

与现有steps使用方法一致

### PR DESCRIPTION
**Pull request by yykoypj**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #35802

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |steps support for circular progress bar          |
| 🇨🇳 Chinese |Progress: circle模式下支持steps           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/progress/index.zh-CN.md [hunk 1]
```diff
@@ -65,13 +66,15 @@ demo:
...
 | --- | --- | --- | --- | --- |
+| steps | 进度条总共步数，传入 object 时，count 指步数，space 指间隔大小。传 number 类型时，space 默认为 2。 | number \| { count: number, space: number } | - | 5.15.4 |
 | strokeColor | 圆形进度条线的色彩，传入 object 时为渐变 | string \| { number%: string } | - | - |
 | strokeWidth | 圆形进度条线的宽度，单位是进度条画布宽度的百分比 | number | 6 | - |
 
 ### `type="dashboard"`
 
 | 属性 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+| steps | 进度条总共步数，传入 object 时，count 指步数，space 指间隔大小。传 number 类型时，space 默认为 2。 | number \| { count: number, space: number } | - | 5.15.4 |
```
Comment (afc163) [inline components/progress/index.zh-CN.md:77]:
```suggestion
| steps | 进度条总共步数，传入 object 时，count 指步数，space 指间隔大小。传 number 类型时，space 默认为 2。 | number \| { count: number, space: number } | - | 5.16.0 |
```

File: components/progress/progress.tsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ export interface ProgressProps extends ProgressAriaProps {
   gapDegree?: number;
   gapPosition?: 'top' | 'bottom' | 'left' | 'right';
   size?: number | [number | string, number] | ProgressSize;
-  steps?: number;
+  steps?: number | { count: number; space: number };
```
Comment (afc163) [inline components/progress/progress.tsx:56]:
```suggestion
  steps?: number | { count: number; gap: number };
```

改成 gap 吧

File: components/progress/index.en-US.md [hunk 3]
```diff
@@ -64,13 +65,15 @@ Properties that shared by all types.
...
 | strokeColor | The color of circular progress, render gradient when passing an object | string \| { number%: string } | - | - |
 | strokeWidth | To set the width of the circular progress, unit: percentage of the canvas width | number | 6 | - |
 
 ### `type="dashboard"`
 
-| Property | Description | Type | Default |
-| --- | --- | --- | --- |
+| Property | Description | Type | Default | Version |
+| --- | --- | --- | --- | --- |
+| steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.15.4 |
```
Comment (li-jia-nan) [inline components/progress/index.en-US.md:76]:
```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.16.0 |
```

File: components/progress/index.en-US.md [hunk 4]
```diff
@@ -32,6 +32,7 @@ If it will take a long time to complete an operation, you can use `Progress` to
 <code src="./demo/linecap.tsx">Stroke Linecap</code>
 <code src="./demo/gradient-line.tsx">Custom line gradient</code>
 <code src="./demo/steps.tsx">Progress bar with steps</code>
+<code src="./demo/circle-steps.tsx">Circular progress bar whit steps</code>
```
Comment (li-jia-nan) [inline components/progress/index.en-US.md:35]:
```suggestion
<code src="./demo/circle-steps.tsx" version="5.16.0">Circular progress bar whit steps</code>
```

File: components/progress/progress.tsx (hunk 2 above)
Comment (afc163) [inline components/progress/progress.tsx:56]: 改成 gap 更好一点？

Comment (li-jia-nan) [reply components/progress/progress.tsx:56]: 是说把 space 改成 gap 吗？我觉得可以

Comment (yykoypj) [reply components/progress/progress.tsx:56]: 好的，搞！

Comment (yykoypj) [reply components/progress/progress.tsx:56]: rc-progress那边已经是space，要改那边吗

Comment (afc163) [reply components/progress/progress.tsx:56]: 一起改了吧。

Comment (li-jia-nan) [inline components/progress/progress.tsx:56]:
```suggestion
  steps?: number | { count?: number; gap?: number };
```

File: components/progress/__tests__/index.test.tsx [hunk 5]
```diff
@@ -404,4 +404,14 @@ describe('Progress', () => {
     expect(progress).toHaveAttribute('aria-labelledby', 'progressLabel');
     expect(progress).toHaveAttribute('aria-valuenow', '90');
   });
+
+  it('circle progress should accept steps', () => {
+    const { container } = render(<Progress percent={70} steps={{ count: 5, space: 5 }} />);
```
Comment (li-jia-nan) [inline components/progress/__tests__/index.test.tsx:409]:
```suggestion
    const { container } = render(<Progress percent={70} steps={{ count: 5, gap: 5 }} />);
```

File: components/progress/demo/circle-steps.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      percent={50}
+      trailColor="rgba(0, 0, 0, 0.06)"
+      strokeWidth={20}
+    />
+    <Progress
+      type="circle"
+      percent={100}
+      steps={{
+        count: 5,
+        space: 12,
```
Comment (li-jia-nan) [inline components/progress/demo/circle-steps.tsx:18]:
```suggestion
        gap: 12,
```

File: components/progress/index.en-US.md [hunk 7]
```diff
@@ -64,13 +65,15 @@ Properties that shared by all types.
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
+| steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.16.0 |
```
Comment (li-jia-nan) [inline components/progress/index.en-US.md:68]:
```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count?: number, gap?: number } | - | 5.16.0 |
```

File: components/progress/index.en-US.md [hunk 8]
```diff
@@ -64,13 +65,15 @@ Properties that shared by all types.
...
 | strokeColor | The color of circular progress, render gradient when passing an object | string \| { number%: string } | - | - |
 | strokeWidth | To set the width of the circular progress, unit: percentage of the canvas width | number | 6 | - |
 
 ### `type="dashboard"`
 
-| Property | Description | Type | Default |
-| --- | --- | --- | --- |
+| Property | Description | Type | Default | Version |
+| --- | --- | --- | --- | --- |
+| steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.16.0 |
```
Comment (li-jia-nan) [inline components/progress/index.en-US.md:76]:
```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count?: number, gap?: number } | - | 5.16.0 |
```

File: components/progress/index.zh-CN.md [hunk 9]
```diff
@@ -65,13 +66,15 @@ demo:
 
 | 属性 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+| steps | 进度条总共步数，传入 object 时，count 指步数，space 指间隔大小。传 number 类型时，space 默认为 2。 | number \| { count: number, space: number } | - | 5.16.0 |
```
Comment (li-jia-nan) [inline components/progress/index.zh-CN.md:69]:
```suggestion
| steps | 进度条总共步数，传入 object 时，count 指步数，gap 指间隔大小。传 number 类型时，gap 默认为 2。 | number \| { count?: number, gap?: number } | - | 5.16.0 |
```

File: components/progress/index.zh-CN.md [hunk 10]
```diff
@@ -65,13 +66,15 @@ demo:
...
 | strokeColor | 圆形进度条线的色彩，传入 object 时为渐变 | string \| { number%: string } | - | - |
 | strokeWidth | 圆形进度条线的宽度，单位是进度条画布宽度的百分比 | number | 6 | - |
 
 ### `type="dashboard"`
 
 | 属性 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+| steps | 进度条总共步数，传入 object 时，count 指步数，space 指间隔大小。传 number 类型时，space 默认为 2。 | number \| { count: number, space: number } | - | 5.16.0 |
```
Comment (li-jia-nan) [inline components/progress/index.zh-CN.md:77]:
```suggestion
| steps | 进度条总共步数，传入 object 时，count 指步数，gap 指间隔大小。传 number 类型时，gap 默认为 2。 | number \| { count?: number, gap?: number } | - | 5.16.0 |
```

**Comment by yykoypj** [comment]
@li-jia-nan 大佬要不把rc-component/progress也改了？

**Comment by li-jia-nan** [comment]
> @li-jia-nan 大佬要不把rc-component/progress也改了？

OK

File: components/progress/progress.tsx [hunk 11]
```diff
@@ -53,7 +53,7 @@ export interface ProgressProps extends ProgressAriaProps {
   gapDegree?: number;
   gapPosition?: 'top' | 'bottom' | 'left' | 'right';
   size?: number | [number | string, number] | ProgressSize;
-  steps?: number;
+  steps?: number | { count?: number; gap?: number };
```
Comment (li-jia-nan) [inline components/progress/progress.tsx:56]:
```suggestion
  steps?: number | { count: number; gap: number };
```

File: components/progress/index.zh-CN.md [hunk 12]
```diff
@@ -65,13 +66,15 @@ demo:
...
 | --- | --- | --- | --- | --- |
+| steps | 进度条总共步数，传入 object 时，count 指步数，gap 指间隔大小。传 number 类型时，gap 默认为 2。 | number \| { count?: number, gap?: number } | - | 5.16.0 |
 | strokeColor | 圆形进度条线的色彩，传入 object 时为渐变 | string \| { number%: string } | - | - |
 | strokeWidth | 圆形进度条线的宽度，单位是进度条画布宽度的百分比 | number | 6 | - |
 
 ### `type="dashboard"`
 
 | 属性 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+| steps | 进度条总共步数，传入 object 时，count 指步数，gap 指间隔大小。传 number 类型时，gap 默认为 2。 | number \| { count?: number, gap?: number } | - | 5.16.0 |
```
Comment (li-jia-nan) [inline components/progress/index.zh-CN.md:77]:
```suggestion
| steps | 进度条总共步数，传入 object 时，count 指步数，gap 指间隔大小。传 number 类型时，gap 默认为 2。 | number \| { count: number, gap: number } | - | 5.16.0 |
```

File: components/progress/index.zh-CN.md [hunk 13]
```diff
@@ -65,13 +66,15 @@ demo:
...
```
Comment (li-jia-nan) [inline components/progress/index.zh-CN.md:69]:
```suggestion
| steps | 进度条总共步数，传入 object 时，count 指步数，gap 指间隔大小。传 number 类型时，gap 默认为 2。 | number \| { count: number, gap: number } | - | 5.16.0 |
```

File: components/progress/index.en-US.md [hunk 14]
```diff
@@ -64,13 +65,15 @@ Properties that shared by all types.
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
+| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count?: number, gap?: number } | - | 5.16.0 |
```
Comment (li-jia-nan) [inline components/progress/index.en-US.md:68]:
```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count: number, gap: number } | - | 5.16.0 |
```

File: components/progress/index.en-US.md [hunk 15]
```diff
@@ -64,13 +65,15 @@ Properties that shared by all types.
...
 | strokeColor | The color of circular progress, render gradient when passing an object | string \| { number%: string } | - | - |
 | strokeWidth | To set the width of the circular progress, unit: percentage of the canvas width | number | 6 | - |
 
 ### `type="dashboard"`
 
-| Property | Description | Type | Default |
-| --- | --- | --- | --- |
+| Property | Description | Type | Default | Version |
+| --- | --- | --- | --- | --- |
+| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count?: number, gap?: number } | - | 5.16.0 |
```
Comment (li-jia-nan) [inline components/progress/index.en-US.md:76]:
```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count: number, gap: number } | - | 5.16.0 |
```

**Comment by li-jia-nan** [comment]
@afc163 求发版，未发布的 api，应该发 patch 就行了
- https://github.com/react-component/progress/pull/275

**Comment by afc163** [comment]
https://github.com/react-component/progress/releases/tag/v4.0.0

**Review by li-jia-nan — DISMISSED** [review]

File: components/progress/demo/circle-steps.md [hunk 16]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+步骤进度圈，支持颜色分段展示
```
Comment (afc163) [inline components/progress/demo/circle-steps.md:3]:
```suggestion
步骤进度圈，支持颜色分段展示。
```

File: components/progress/demo/circle-steps.tsx [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+      trailColor="rgba(0, 0, 0, 0.06)"
+      strokeWidth={20}
+    />
+    <Progress
+      type="circle"
+      percent={100}
+      steps={{ count: 5, gap: 12 }}
+      trailColor="rgba(0, 0, 0, 0.06)"
+      strokeWidth={20}
+    />
```
Comment (afc163) [inline components/progress/demo/circle-steps.tsx:19]: 这两个 demo 不太协调好看，搞两个真实的设计效果出来？

Comment (li-jia-nan) [reply components/progress/demo/circle-steps.tsx:19]: @yykoypj 你来改一下？

**Review by li-jia-nan — APPROVED** [review]

File: components/progress/demo/circle-steps.tsx (hunk 17 above)
Comment (li-jia-nan) [reply components/progress/demo/circle-steps.tsx:19]: @afc163 要不先合了，我单独开个 pr 改一下 demo ？

**Comment by afc163** [comment]
## size-limit report 📦 
| Path                              | Size                  |
| --------------------------------- | --------------------- |
| `./dist/antd.min.js`              | 335.12 KB (-2 B 🔽)   |
| `./dist/antd-with-locales.min.js` | 381.74 KB (-107 B 🔽) |

**Review by afc163 — APPROVED** [review]

File: components/progress/demo/circle-steps.tsx (hunk 17 above)
Comment (li-jia-nan) [reply components/progress/demo/circle-steps.tsx:19]: 加了两个滑动条：
- #48052

**Pull request merged**